### PR TITLE
feat: video generation plugin

### PR DESCRIPTION
**Pull request by dorianjanezic**
Relates to:
video generation

Risks
Low - Changes are contained within the video generation plugin and don't affect other functionality. Main risk is failed video generation or attachment, which is handled gracefully with error messages.

Background
What does this PR do?
Fixes video generation plugin to properly handle the Luma AI API response and correctly attach generated videos to Discord messages.

What kind of change is this?
Plugin

Why are we doing this? Any context or related work?
adding video gen feature

Documentation changes needed?
Yes, adding LUMA LABS API to env.example

Testing
As a user, send a message to generate a video:
Send "Generate a video of a sunset"
Verify bot responds with "I'll generate a video based on your prompt"
Wait for video generation (about 1-2 minutes)
Verify bot sends back a message with the video attached

videoGenerationPlugin should be added to agent runtime & to package.json

### CODE REVIEW

**Comment by dorianjanezic** [comment]
@lalalune hey, all good with this plugin?

**Comment by monilpat** [comment]
Thanks for working on this - Please update the PR title :)

File: packages/plugin-video-generation/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    HandlerCallback,
+    IAgentRuntime,
+    Memory,
+    Plugin,
+    State,
+} from "@ai16z/eliza/src/types.ts";
+import fs from "fs";
+
+const generateVideo = async (prompt: string, runtime: IAgentRuntime) => {
+    const API_URL = 'https://api.lumalabs.ai/dream-machine/v1/generations';
```
Comment (monilpat) [inline packages/plugin-video-generation/src/index.ts:13]: Can we move this to a constants file

File: packages/plugin-video-generation/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+        callback: HandlerCallback
+    ) => {
+        elizaLogger.log("Video generation request:", message);
+
+        // Clean up the prompt by removing mentions and commands
+        let videoPrompt = message.content.text
+            .replace(/<@\d+>/g, '')  // Remove mentions
+            .replace(/generate video|create video|make video|render video/gi, '')  // Remove commands
+            .trim();
+
```
Comment (monilpat) [inline packages/plugin-video-generation/src/index.ts:132]: Why do we need to clean it up? Thanks for clarifying!

**Comment by monilpat** [comment]
Can we add a test for this new plugin? As well as a screengrab of successful video generation? Thanks so much!

File: packages/plugin-video-generation/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    IAgentRuntime,
+    Memory,
+    Plugin,
+    State,
+} from "@ai16z/eliza/src/types.ts";
+import fs from "fs";
+
+const generateVideo = async (prompt: string, runtime: IAgentRuntime) => {
+    const API_URL = 'https://api.lumalabs.ai/dream-machine/v1/generations';
+    const API_KEY = runtime.getSetting("LUMA_API_KEY");
```
Comment (monilpat) [inline packages/plugin-video-generation/src/index.ts:14]: Can we update the .env.example file to include the LUMA_API_KEY?

**Review by monilpat — CHANGES_REQUESTED** [review]
Thanks for working on this! Great feature and implementation! Please address the comments then good to go!

**Comment by dorianjanezic** [comment]
hey, created constants file and videoGeneration test. 

Screenshot 2024-11-18 at 11 58 23

**Pull request merged**